genic/zeldovich.c

```diff
@@ -420,7 +420,13 @@ pmic_fill_gaussian_gadget(PM * pm, double * delta_k, int seed, int unitary)
                 /* we want two numbers that are of std ~ 1/sqrt(2) */
                 ampl = sqrt(- log(ampl));
 
-                if (unitary) ampl = 1.0; /* cos and sin gives 1/sqrt(2)*/
+                if (UnitaryAmplitude) ampl = 1.0; /* cos and sin gives 1/sqrt(2)*/
```

Comment:
UnitaryAmplitude and InvertPhase are a global variables, in my version I passed them as parameters changing the following lines:

static void gaussian_fill(PetaPMRegion * region, pfft_complex * rho_k, int unitaryAmplitude, int invertPhase)

gaussian_fill(petapm_get_fourier_region(), petapm_get_rho_k(), UnitaryAmplitude, InvertPhase);

pmic_fill_gaussian_gadget(PM * pm, double * delta_k, int seed, int unitaryAmplitude, int invertPhase)

if (unitaryAmplitude) ampl = 1.0; /* cos and sin gives 1/sqrt(2)*/

if (invertPhase) phase = phase + M_PI;  /* invert the sign of the delta */

Reply:
I set UnitaryAmplitude and InvertPhase in the paramfile for mp-genic. Are you suggesting a change? I think I'm not understanding your comment

Reply:
There is an unused parameter `unitary` in the argument list of this function. You can either get rid of that; or you can make sure no global variables is referenced in this function by passing both in as parameters, like Elena's version. Since this piece of code has been somewhat shared between FastPM[1], pmesh[2], and MP-Gadget, the no-global variable approach will produce slightly less headache in the future. But it's really up to you to decide which way to go.

[1] https://github.com/rainwoodman/fastpm/blob/master/libfastpm/initialcondition.c#L141
[2] https://github.com/rainwoodman/pmesh/blob/master/pmesh/_whitenoise_generics.h#L2